prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -318,6 +328,43 @@ object Parser extends Expression {
     }
   }
 
+  def parseMetadataQuery(query: String): Expression = {
```

Comment:
I don't think this method and `parseMetadataLabelValuesQuery` method is needed.,  The regular parsing method can parse vector and filter definitions into Expressions.  The only thing which is really needed is the `metadataQueryToLogicalPlan` method as that requires different logical plan translations.